Add .test tld

kmsheng opened the pull request:
https://en.wikipedia.org/wiki/.test

This PR
 - adds `.test` to TLD list
 - updates build/*
 - adds `.test` test case to b_integration/basic.test.ts

kmsheng committed:
add .test TLD

kmsheng committed:
update dist folder

kmsheng committed:
add .test domain to example.urls.valid

alexcorvi commented:
Hello, you've manually changed the [src/tlds.ts](https://github.com/alexcorvi/anchorme.js/pull/73/files#diff-dcaca78d70681e8b59c52e4f02d0de80) file, this file get updated on-demand by running [`npm run update-tlds`](https://github.com/alexcorvi/anchorme.js/blob/gh-pages/package.json#L12).
So I'd suggest adding the "test" TLD to the updating function [here](https://github.com/alexcorvi/anchorme.js/blob/gh-pages/build/tlds/update.ts)

kmsheng committed:
Revert "add .test TLD"

This reverts commit 8bd91c1989f244715bfc1bc55095c9b016982127.

kmsheng committed:
add const localTLDs to build/tlds/update.ts

kmsheng committed:
update tlds.ts by running "npm run update-tlds"

kmsheng committed:
check duplication when merging with local TLDs

kmsheng commented:
@alexcorvi You're right.
 I've updated `build/tlds/update.ts`, so it can merge local TLDs with remote TLDs.
Also I've added duplication checking, so if `TEST` were added in https://data.iana.org/TLD/tlds-alpha-by-domain.txt someday, this program will still be fine.
